cmd/cluster-agent/custommetrics/server.go

```diff
@@ -86,7 +87,7 @@ func (a *DatadogMetricsAdapter) makeProviderOrDie() (provider.ExternalMetricsPro
 		return nil, err
 	}
 
-	return custommetrics.NewDatadogProvider(client, mapper, store), nil
+	return custommetrics.NewDatadogProvider(context.Background(), client, mapper, store), nil
```

Comment:
What do you think if the context comes from `func StartServer(ctx context.Context)` then propagated to `DatadogMetricsAdapter.makeProviderOrDie(ctx context.Context)` ?

Reply:
That's not a bad idea - I was not sure how far I wanted the context to be from. In a follow up PR I want to implement the retry on the configmap like the leader election. If it's deleted, we can recreate it on the fly without terminating the server. 

For now, I think serving 503 is the best.